main.py

```
@@ -78,5 +105,331 @@ async def dispatch(self, request: Request, call_next):
+                    "error": f"Failed to get operation status: {str(e)}",
+                    "operation_name": operation_name,
+                    "user_id": operation_data['user_id'],
+                    "session_id": operation_data['session_id'],
+                    "created_at": operation_data['created_at']
+                }
+
+    except HTTPException as he:
+        raise he
+    except Exception as e:
+        logger.error(f"Error checking video status: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Unexpected error: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)